pkg/controller/webhook/federatedcluster_webhook.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1/validation"
+)
+
+type FederatedClusterValidationHook struct {
+	client dynamic.ResourceInterface
+
+	lock        sync.RWMutex
+	initialized bool
+}
+
+func (a *FederatedClusterValidationHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	return schema.GroupVersionResource{
+			Group:    "admission.core.federation.k8s.io",
+			Version:  "v1alpha1",
+			Resource: "federatedclusters",
```

Comment:
Some of the strings, for example the resource plural name here could be made as constants for usage everywhere.

Reply:
Yep there's a lot of room for refactoring of this code after the base functionality is working.